backend/app/Http/Controllers/CalendarActivityController.php

```diff
@@ -4,62 +4,69 @@
 
 use App\Models\CalendarActivity;
 use Illuminate\Http\Request;
+use App\Models\Journey;
+use Illuminate\Support\Facades\Gate;
 
 class CalendarActivityController extends Controller
 {
     /**
-     * Display a listing of the resource.
+     * Add an activity to the calendar of the journey.
      */
-    public function index()
+    public function store(Request $request, $journey, $activity)
     {
-        //
-    }
+        $journey = Journey::findOrFail($journey);
+        Gate::authorize("journeyGuide", $journey);
 
-    /**
-     * Show the form for creating a new resource.
-     */
-    public function create()
-    {
-        //
-    }
+        $activity = $journey->activities()->findOrFail($activity);
 
-    /**
-     * Store a newly created resource in storage.
-     */
-    public function store(Request $request)
-    {
-        //
-    }
+        $validated = $request->validate([
+            "start" => "required|date",
+            "end" => "required|date",
+        ]);
 
-    /**
-     * Display the specified resource.
-     */
-    public function show(CalendarActivity $calendarActivity)
-    {
-        //
-    }
+        $calendarActivity = new CalendarActivity($validated);
+        $calendarActivity->activity_id = $activity->id;
 
-    /**
-     * Show the form for editing the specified resource.
-     */
-    public function edit(CalendarActivity $calendarActivity)
-    {
-        //
+        $calendarActivity->save();
+
+        return response()->json($calendarActivity, 201);
     }
 
     /**
      * Update the specified resource in storage.
      */
-    public function update(Request $request, CalendarActivity $calendarActivity)
-    {
-        //
+    public function update(
+        Request $request,
+        $journey,
+        $activity,
```

Comment:
looks unused @Mathematti

Reply:
but idk if we alr use this function

(if yes, you could also update the comment and the comment for the destroy function where activity is also unused)